### PR TITLE
USHIFT-3779: Check if nginx is running in scenario.sh and build_images.sh

### PR DESCRIPTION
**Pull request by vanhalenar**

### CODE REVIEW

File: test/bin/build_images.sh [hunk 1]
```diff
@@ -706,6 +706,11 @@ fi
 
 trap 'osbuild_logs' EXIT
 
+# Check if webserver is running
+if ! ps aux | grep nginx | grep -v grep > /dev/null; then
```
Comment (ggiguash) [inline test/bin/build_images.sh:710]: This might be a more effective way to check:
```
if [ $(pgrep -cx nginx) -eq 0 ] ; then
fi
```

Comment (vanhalenar) [reply test/bin/build_images.sh:710]: yes, I just ran shellcheck and it suggested the same thing so I'm gonna do that, thanks :)

File: test/bin/scenario.sh [hunk 2]
```diff
@@ -930,6 +930,19 @@ load_subscription_manager_plugin() {
...
 
+# Check if dependencies are running, and if not, start them
+#   - nginx server
+#   - registry mirror
+check_dependencies() {
+    if [ $(pgrep -cx nginx) -eq 0 ] ; then
+        "${TESTDIR}/bin/manage_webserver.sh" "start"
+    fi
+
+    if ! podman container inspect microshift-local-registry > /dev/null 2>&1; then
```
Comment (ggiguash) [inline test/bin/scenario.sh:941]:
```suggestion
    if ! podman container inspect microshift-local-registry &> /dev/null ; then
```

Comment (vanhalenar) [reply test/bin/scenario.sh]: done

File: test/bin/scenario.sh [hunk 3]
```diff
@@ -930,6 +932,19 @@ load_subscription_manager_plugin() {
...
 
+# Check if dependencies are running, and if not, start them
+#   - nginx server
+#   - registry mirror
+check_dependencies() {
+    if [ $(pgrep -cx nginx) -eq 0 ] ; then
+        "${TESTDIR}/bin/manage_webserver.sh" "start"
+    fi
+
+    if ! podman ps | grep microshift-local-registry &> /dev/null ; then
```
Comment (ggiguash) [inline test/bin/scenario.sh:943]:
```suggestion
    if ! podman ps --format '{{.Names}}' | grep -q ^microshift-local-registry  ; then
```

File: test/bin/pyutils/build_bootc_images.py [hunk 4]
```diff
@@ -523,7 +523,8 @@ def main():
             raise Exception(f"The input directory '{dir2process}' does not exist")
         # Make sure the local RPM repository exists
         if not os.path.isdir(LOCAL_REPO):
-            raise Exception("Run create_local_repo.sh before building images")
+            build_rpms = f"{SCRIPTDIR}/build_rpms.sh"
+            common.run_command(build_rpms)
```
Comment (ggiguash) [inline test/bin/pyutils/build_bootc_images.py:527]:
```suggestion
            common.run_command(f"{SCRIPTDIR}/build_rpms.sh")
```

File: test/bin/scenario.sh [hunk 5]
```diff
@@ -801,11 +802,12 @@ run_tests() {
     record_junit "${vmname}" "robot_framework_installed" "OK"
 
     # Make sure oc command is available
-    if ! command -v oc &> /dev/null
-    then
-        error "OpenShift Client package not installed, install it with ${ROOTDIR}/scripts/fetch_tools.sh oc"
-        record_junit "${vmname}" "oc_installed" "FAILED"
-        exit 1
+    if ! command -v oc &> /dev/null ; then
+        echo "OpenShift Client package not installed, installing with ${ROOTDIR}/scripts/fetch_tools.sh oc"
```
Comment (ggiguash) [inline test/bin/scenario.sh:806]: Let's remove the echo to be consistent with other places?

File: test/bin/scenario.sh (hunk 3 above)
Comment (vanhalenar) [reply test/bin/scenario.sh]: Done

File: test/bin/pyutils/build_bootc_images.py (hunk 4 above)
Comment (vanhalenar) [reply test/bin/pyutils/build_bootc_images.py]: Done

File: test/bin/scenario.sh [hunk 6]
```diff
@@ -787,9 +786,11 @@ run_tests() {
     echo "Running tests with $# args" "$@"
 
     if [ ! -d "${RF_VENV}" ]; then
-        error "RF_VENV (${RF_VENV}) does not exist, create it with: ${ROOTDIR}/scripts/fetch_tools.sh robotframework"
-        record_junit "${vmname}" "robot_framework_environment" "FAILED"
-        exit 1
+        echo "RF_VENV (${RF_VENV}) does not exist, running ${ROOTDIR}/scripts/fetch_tools.sh robotframework"
```
[the remark this answers is not in the document]
Comment (vanhalenar) [reply test/bin/scenario.sh]: Done

File: test/bin/scenario.sh (hunk 5 above)
Comment (vanhalenar) [reply test/bin/scenario.sh]: Done

**Pull request merged**